viperx/output.go

```diff
@@ -76,3 +81,22 @@ func jsonschemaFormatError(e *jsonschema.ValidationError) (string, string) {
 
 	return pointer, message
 }
+
+const permOwnerRW = 0600
+
+var lastDumpedConfig = -1
+
+func getDumpFileName(i int) string {
+	return fmt.Sprintf("config-%08d.yml", i)
+}
+
+func sensitiveDumpAllValues(dir string) error {
+	configContent, err := yaml.Marshal(viper.AllSettings())
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	lastDumpedConfig++
```

Comment:
This is racy because it may be called by several goroutines simoultaneously. Use `atomic` if you want a global counter: https://golang.org/pkg/sync/atomic/ 

Why not just use the unix nano timestamp? `time.Now().UnixNano()` which is most likely unique enough?